extension/basicauth/authdata.go

```diff
@@ -0,0 +1,77 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package basicauth // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/basicauth"
+
+import (
+	"encoding/base64"
+	"errors"
+	"strings"
+
+	"go.opentelemetry.io/collector/client"
+)
+
+var (
+	errInvalidSchemePrefix = errors.New("invalid authorization scheme prefix")
+	errInvalidFormat       = errors.New("invalid format")
+)
+
+// See: https://github.com/golang/go/blob/1a8b4e05b1ff7a52c6d40fad73bcad612168d094/src/net/http/request.go#L950
+func parseBasicAuth(auth string) (*authData, error) {
+	const prefix = "Basic "
+	if len(auth) < len(prefix) || !strings.EqualFold(auth[:len(prefix)], prefix) {
+		return nil, errInvalidSchemePrefix
+	}
+
+	encoded := auth[len(prefix):]
+	decodedBytes, err := base64.StdEncoding.DecodeString(encoded)
+	if err != nil {
+		return nil, errInvalidFormat
+	}
+	decoded := string(decodedBytes)
+
+	si := strings.IndexByte(decoded, ':')
+	if si < 0 {
+		return nil, errInvalidFormat
+	}
+
+	return &authData{
+		username: decoded[:si],
+		password: decoded[si+1:],
+		raw:      encoded,
+	}, nil
+}
+
+var _ client.AuthData = (*authData)(nil)
```

Comment:
This is server auth, is it ok to be `client.AuthData`?

Reply:
No worries if this is unrelated to this PR just wanted to point out this is confusing to me, and in the other function which is also using the `client` package

Reply:
I took the approach taken in `oidc` extension. I believe the semantics here is "calling client authentication data".